src/ivc/incrementally_verifiable_computation.rs

```diff
@@ -71,19 +79,233 @@ where
+            let mut layouter =
+                SingleChipLayouter::<'_, C1::Scalar, _>::new(&mut primary_td, vec![])?;
+
+            let primary_assigned_z0: [_; A1] = layouter.assign_region(
+                || "assigned_z0_primary",
+                |region| {
+                    primary_config
+                        .main_gate_config
+                        .advice_cycle_assigner()
+                        .assign_all_advice(
+                            &mut RegionCtx::new(region, 0),
+                            || "z0_primary",
+                            z0_primary.iter().copied(),
+                        )
+                        .map(|inp| inp.try_into().unwrap())
+                },
+            )?;
+            debug!("primary z0 assigned");
+            debug!("start primary synthesize");
+
+            let StepSynthesisResult {
+                z_output: primary_assigned_z_output,
+                // Not used in zero step, only in verify-step, when folding is over
+                output_hash: _primary_output_hash,
+                // Not used in zero step, only in verify-step, when folding is over
+                X1: _primary_X1,
+            } = primary.synthesize(
+                primary_config,
+                &mut layouter,
+                step_circuit::StepInputs {
+                    step_public_params: &pp.primary.params,
+                    public_params_hash: primary_public_params_hash,
+                    step: C1::Scalar::ZERO,
+                    z_0: primary_assigned_z0.clone(),
+                    z_i: primary_assigned_z0.clone(),
+                    // Can be any
+                    U: PlonkInstance::default().to_relax(),
+                    // Can be any
+                    u: PlonkInstance::default(),
+                    cross_term_commits: vec![C2::identity(); primary_cross_term_commits_len],
+                },
+            )?;
+
+            debug!("start primary td postpone");
+            primary_td.postpone_assembly();
+            debug!("primary synthesized, start sps");
+
+            let (primary_plonk_instance, primary_plonk_witness) = primary_td.run_sps_protocol(
+                pp.primary.ck,
+                &mut RP2::OffCircuit::new(pp.secondary.params.ro_constant.clone()),
+                primary_td.plonk_structure().unwrap().num_challenges,
+            )?;
+
+            (
+                StepCircuitContext::<A1, C1, SC1> {
+                    step_circuit: primary,
+                    relaxed_trace: RelaxedPlonkTrace {
+                        U: primary_plonk_instance.to_relax(),
+                        W: primary_plonk_witness.to_relax(pp.primary.td_k() as usize),
+                    },
+                    z_0: primary_assigned_z0,
+                    z_i: primary_assigned_z_output,
+                },
+                primary_plonk_instance,
+            )
+        };
+        debug!("primary ctx ready");
+
+        let (secondary_ctx, secondary_trace) = {
```

Comment:
`secondary_trace` here the naming is not aligned with primary case `primary_plonk_instance`.